storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/impl/ChainImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.pushtorefresh.storio.sqlite.impl;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.pushtorefresh.storio.operations.PreparedOperation;
+import com.pushtorefresh.storio.sqlite.Interceptor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ChainImpl implements Interceptor.Chain {
+
+    @NonNull
+    public static Interceptor.Chain buildChain(
+            @NonNull List<Interceptor> registeredInterceptors,
+            @NonNull Interceptor realInterceptor
+    ) {
+        final List<Interceptor> interceptors = new ArrayList<Interceptor>(registeredInterceptors.size() + 1);
+        interceptors.addAll(registeredInterceptors);
+        interceptors.add(realInterceptor);
+
+        return new ChainImpl(interceptors, 0);
+    }
+
+    @NonNull
+    private final List<Interceptor> interceptors;
+
+    @NonNull
+    private final int index;
+
+    private int calls;
```

Comment:
Can be changed to just a `boolean` flag since according to the code we don't really care about actual number of calls.